src/Angor/Avalonia/Angor.Projects/Infrastructure/Interfaces/IInvestmentRepository.cs

```diff
@@ -0,0 +1,10 @@
+using Angor.Projects.Domain;
+using CSharpFunctionalExtensions;
+
+namespace Angor.Projects.Infrastructure.Interfaces;
+
+public interface IInvestmentRepository
+{
+    Task<Result> Save(Investment investment);
+    Task<Result<IEnumerable<Investment>>> Get(ProjectId projectId);
```

Comment:
A founder can only invest once in a project, this would reflect better an endpoint that gets all investment from the indexer service

Reply:
Do you mean to change the Get method to Task<Result<IEnumerable<Investment>>> GetAll()?